spock-core/src/main/java/org/spockframework/mock/runtime/GroovyMockFactory.java

```diff
@@ -131,10 +129,31 @@ private boolean isIsolatedOrHasMetaClassRegistryReadWriteLock(Specification spec
     return feature.getExclusiveResources().contains(META_CLASS_REGISTRY_RW);
   }
 
+  private Object createProxyObjectForAbstractGlobalMock(IMockConfiguration configuration, Specification specification, GroovyMockMetaClass mockMetaClass) {
+    IProxyBasedMockInterceptor mockInterceptor = new GroovyMockInterceptor(configuration, specification, mockMetaClass);
+    //Issue #464: We need to create a subclass mock here to be able to instantiate the abstract class.
+    Object proxy = createMockObject(configuration, specification, mockInterceptor, mockMetaClass);
+    //We need also to replace the metaClass of the Proxy Object to ensure correct usage of the mock object.
+    GroovyRuntimeUtil.setMetaClass(proxy, mockMetaClass);
+    return proxy;
+  }
+
+  private Object createMockObject(IMockConfiguration configuration, Specification specification, IProxyBasedMockInterceptor mockInterceptor, GroovyMockMetaClass mockMetaClass) {
+    IMockMaker.IMockCreationSettings mockCreationSettings = MockCreationSettings.settingsFromMockConfiguration(configuration,
+      mockInterceptor,
+      specification.getClass().getClassLoader());
```

Comment:
Is that the correct classloader? Shouldn't we use the one from the abstract type?

Reply:
The old code and also the `JavaMockFactory` is using the `Specification` classloader. 
I also think it is the better one, because the `Specification` classloader is capable of seeing the `Mock` class and also the Spock classes, which can be an issue in in an `OSGi` env. 

If we use the `Classloader` of the Mock class, it could happen of e.g. interfaces that the Spock class `ISpockMockObject` would not be visible by the `Classloader` of the abstract class, if the abstract class came from a bundle, which did not import Spock.

Reply:
Ohh that leads to another question, what if the user passes additional interfaces from different classloaders to the `JavaProxyMockMaker`, I guess that will currently fail. We probably need to do something like byte-buddy and mockito do to calculate if all classes are accessible, if not they construct a new delegating classloader, which aggregates the different classloaders to make that happen. Or maybe not to duplicate the code, if we detect that case in `JavaProxyMockMaker` switch over to the `ByteBuddyMockMaker`, which already supports that.

I will will out, if this fails in another branch and open up a PR, if that is an issue.